js/main.js

```diff
@@ -1,2 +1,31 @@
 /* global data */
 /* exported data */
+
+// Event listener for image url
+var $urlImage = document.querySelector('.url-image');
+var $url = document.querySelector('#entry-form-photo-url');
+
+function updateEntryImage(event) {
+  var urlInputValue = event.target.value;
+  $urlImage.setAttribute('src', urlInputValue);
+}
+
+$url.addEventListener('input', updateEntryImage);
+
+// Event listener for submit event
+var $newEntryForm = document.querySelector('.new-entry-form');
+
+function saveNewEntry(event) {
+  event.preventDefault();
+  var entryInputs = {};
+  entryInputs.title = $newEntryForm.elements.title.value;
+  entryInputs.photo = $newEntryForm.elements.photo.value;
+  entryInputs.note = $newEntryForm.elements.note.value;
+  entryInputs.entryId = data.nextEntryId;
+  data.nextEntryId++;
+  data.entries.push(entryInputs);
```

Comment:
The data needs to be added to the beginning of the array, not the end. You will need a different method than `push`